thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/util/ThirdEyeUtils.java

```diff
@@ -122,51 +130,64 @@ public static String getSortedFiltersFromJson(String filterJson) {
     return sortedFilters;
   }
 
-  /** Returns the collection name for the provided alias, or the input string if it is not an alias. */
-  public static String getCollectionFromAlias(String alias) throws ExecutionException {
-    String collectionName = alias;
-    try {
-      collectionName = CACHE_REGISTRY_INSTANCE.getCollectionAliasCache().get(alias);
-    } catch (InvalidCacheLoadException e) {
-      LOG.debug("No collection name for alias {}", alias);
+  public static TimeSpec getTimeSpecFromDatasetConfig(DatasetConfigDTO datasetConfig) {
+    TimeSpec timespec = new TimeSpec(datasetConfig.getTimeColumn(),
+        new TimeGranularity(datasetConfig.getTimeDuration(), datasetConfig.getTimeUnit()),
+        datasetConfig.getTimeFormat());
+    return timespec;
+  }
+
+  public static MetricExpression getMetricExpressionFromMetricConfig(MetricConfigDTO metricConfig) {
+    MetricExpression metricExpression = new MetricExpression();
+    metricExpression.setExpressionName(metricConfig.getName());
+    if (metricConfig.isDerived()) {
+      metricExpression.setExpression(metricConfig.getDerivedMetricExpression());
+    } else {
+      metricExpression.setExpression(MetricConfigBean.DERIVED_METRIC_ID_PREFIX + metricConfig.getId());
     }
-    return collectionName;
+    return metricExpression;
   }
 
-  /** Returns the alias for the provided collection, or the input string if no alias exists. */
-  public static String getAliasFromCollection(String collection) throws ExecutionException {
-    String result = collection;
-    try {
-      CollectionConfig collectionConfig =
-          CACHE_REGISTRY_INSTANCE.getCollectionConfigCache().get(collection);
-      String configAlias = collectionConfig.getCollectionAlias();
-      if (StringUtils.isNotBlank(configAlias)) {
-        result = configAlias;
+  public static String substituteMetricIdsForMetrics(String metricExpression, String dataset) {
+    MetricConfigManager metricConfigDAO = DAO_REGISTRY.getMetricConfigDAO();
```

Comment:
better to write a parser instead of looking for the occurrence of every metric, add a TODO:. It's ok for now but when the number of metrics increases and we start supporting expressions across datasets, this will impact performance.